h/static/scripts/directive/excerpt.coffee

```diff
@@ -0,0 +1,35 @@
+###*
+# @ngdoc directive
+# @name excerpt
+# @restrict C
+# @description Checks to see if text is overflowing its container.
+# If so, it prepends/appends expanders/collapsers. Note, to work
+# the element needs a max height.
+###
+module.exports = [ '$timeout', ($timeout) ->
+  link: (scope, elem, attr, ctrl) ->
+    # Check for excerpts if an annotation has been created / edited.
+    scope.$watch (-> scope.vm.editing), (editing) ->
```

Comment:
Both this and the the `threadToggleCollapse` event are hints that this isn't really an independent directive. That means that the logic of whether or not to display the "More"/"Less" collapsers is rather complicated.

What I think we should find a way to do is include the `excerpt` directive *inside* the relevant `annotation` templates, and keep the conditional that says ("only if I'm not editing") in there and independent of the `excerpt` directive.